### ISSUE TITLE
Protocols with runtime_checkable and modules are not correctly typed

### ISSUE THREAD

**Issue by Zomatree** [microsoft/pyright#1902]
**Describe the bug**
When using isinstance with a module and a protocol, pyright sets the type to `Never` rather than the module.

**To Reproduce**
import a module via `importlib.import_module` and then use `isinstance` on it with a runtime_checkable protocol, the type is not correctly set on the module after the isinstance. instead of being a module it is turned to `Never`.

**Expected behavior**
It should be set to to the module that also implements the protocol

**Screenshots or Code**
![image](https://user-images.githubusercontent.com/39768508/119276304-fc66e600-bc11-11eb-8205-baa132a29025.png)

**VS Code extension or command-line**
`pyright 1.1.136`

**Additional context**
Mypy types this correctly outputting
```
test.py:13: note: Revealed type is 'test.<subclass of "ModuleType" and "ModuleProtocol">'
```

**Comment by erictraut** [comment]
Thanks for the feature request.

You're correct that the `isinstance` type narrowing logic doesn't currently handle the case where the first argument is a module. That's an easy change to make.

Here's [the change](https://github.com/microsoft/pyright/commit/902907b41f41d0ec826b7b941eb176c3c011a27c), which will be included in the next release.

**Comment by erictraut** [comment]
![Screen Shot 2021-05-23 at 2 32 54 PM](https://user-images.githubusercontent.com/7040122/119277316-c440b280-bbd3-11eb-8cfe-681e26653ba8.png)

**Comment by erictraut** [comment]
This is included in pyright 1.1.143, which I just published. It will also be included in the next release of pylance.
